lib/tokens/TokenManager.js

```diff
@@ -32,10 +32,14 @@ class TokenManager {
   _autoRenew(defaultMaxAgeSecs) {
     debug('Auto renewing token now...');
     this._renew().then((response) => {
-      let maxAgeSecs = cookie.parse(response.headers['set-cookie'][0])['Max-Age'] || defaultMaxAgeSecs;
-      let delaySecs = maxAgeSecs / 2;
-      debug(`Renewing token in ${delaySecs} seconds.`);
-      setTimeout(this._autoRenew.bind(this), delaySecs * 1000).unref();
+      if (Array.isArray(response)) {
+        // renewal required multiple requests (i.e. IAM)
+        response = response[response.length - 1];
+      }
+      let maxAgeSecs = cookie.parse(response.headers['set-cookie'][0])['Max-Age'] || 3600000;
```

Comment:
I think likely a bad merge here, since this has removed the configurable `defaultMaxAgeSecs`